src/components/DependabotAlertsTable/DependabotAlertsTable.tsx

```diff
@@ -0,0 +1,225 @@
+
+const getElapsedTime = (date: string) => {
+  return moment(date).format("YYYY-MM-DD");
+};
+
+type Node = {
+  createdAt: string;
+  dismissedAt: string;
+  securityVulnerability: {
+    package: {
+      name: string;
+    };
+    severity: string,
+    advisory: {
+      description: string,
+    },
+    firstPatchedVersion: {
+      identifier: string,
+    }
+  }
+};
+
+type Repository = {
+  vulnerabilityAlerts: {
+    totalCount: number;
+    nodes: Node[];
+  };
+};
+
+type DetailsUrl = {
+  hostname: string;
+  owner: string;
+  repo: string;
+};
+
+type DenseTableProps = {
+  repository: Repository;
+  detailsUrl: DetailsUrl;
+};
+
+const capitalize = (s: string) => {
+  return s.charAt(0).toLocaleUpperCase('en-US') + s.slice(1);
+};
+
+const getDetailsUrl = (packageName: string, detailsUrl: DetailsUrl, dismissedAt: string) => {
+  const status = dismissedAt ? "closed" : "open";
+  const url = `https://${detailsUrl.hostname}/${detailsUrl.owner}/${detailsUrl.repo}/security/dependabot/yarn.lock/${packageName}/${status}`
+  return <Link to={url}>Details</Link>
+};
+
+export const DenseTable: FC<DenseTableProps> = ({ repository, detailsUrl }) => {
+  const { entity } = useEntity();
+  const projectName = useProjectName(entity);
+  const [filteredTableData, setFilteredTableData] = useState<Node[]>([]);
+  const [insightsStatusFilter, setInsightsStatusFilter] = useState<any>('all');
+
+  const columns: TableColumn[] = [
+    { title: 'Name', field: 'name' },
+    { title: 'Created', field: 'createdAt' },
+    { title: 'State', field: 'state' },
+    { title: 'Details', field: 'details' },
+    { title: 'Severity', field: 'severity' },
+    { title: 'Patched Version', field: 'patched_version' },
+  ];
+  const tableData = filteredTableData.length > 0 ? filteredTableData : repository.vulnerabilityAlerts.nodes
+
+  const structuredData = tableData.map(node => {
+    return {
+      createdAt: getElapsedTime(node.createdAt),
+      state: node.dismissedAt ? 'Dismissed' : 'Open',
+      name: node.securityVulnerability.package.name,
+      details: getDetailsUrl(node.securityVulnerability.package.name, detailsUrl, node.dismissedAt),
+      severity: capitalize((node.securityVulnerability.severity).toLowerCase()),
+      patched_version: node.securityVulnerability.firstPatchedVersion.identifier
+    };
+  });
+  return (
+    <Table
+      title={
+        <>
+          <Box display="flex" alignItems="center">
+            <GitHubIcon />
+            <Box mr={1} />
+            <Typography variant="h6">{projectName}</Typography>
+          </Box>
+          <Paper>
+            <Box position="absolute" right={350} top={20}>
+              <ButtonGroup color="primary" aria-label="text primary button group">
+                <Button
+                  color={insightsStatusFilter === 'all' ? 'primary' : 'default'}
+                  onClick={() => {
+                    setInsightsStatusFilter(
```

Comment:
You could pull these three onClick callbacks into a method that takes the string and a filter function in as arguments so there's a bit less duplication here